clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -302,27 +302,20 @@ private long sendProducerData(long now) {
     }
 
     private boolean maybeSendTransactionalRequest(long now) {
-        if (transactionManager.isCompletingTransaction() &&
```

Comment:
Discussed briefly offline. I will probably leave this for future work. There doesn't seem to be a natural way to do this kind of testing at the moment without exposing a lot of internals.